requirements-test.txt

```diff
@@ -2,14 +2,16 @@ pytest==7.4.4 ; python_version == '3.8'
 pytest==8.2.1 ; python_version >= '3.9'
 numpy>=1.19.5 ; python_version <= '3.9'
 numpy>=1.21.6 ; python_version == '3.10'
-numpy>=1.23.5 ; python_version >= '3.11'
+numpy>=1.23.5 ; python_version == '3.11'
+numpy==2.0.0rc2 ; python_version >= '3.12'
 scikit-learn==1.2.2 ; python_version == '3.8'
 scikit-learn==1.4.2 ; python_version >= '3.9'
 pandas==2.0.3 ; python_version == '3.8'
-pandas==2.1.3 ; python_version >= '3.9'
+pandas==2.1.3 ; python_version >= '3.9' and python_version < '3.11'
+pandas==2.2.2 ; python_version >= '3.11'
 xgboost==2.0.3
 lightgbm==4.3.0
-catboost==1.2.5
+catboost==1.2.5 ; python_version < '3.12' # TODO: Remove 3.12 condition when catboost supports numpy 2.0
```

Comment:
Can/should we split this? CatBoost required only for FTI testing, so it's not required dependency for scikit

Reply:
Currently requirements-test contains all dependencies for public and internal CI. It could be split into "deps required for using the library after building" (ie scikit-learn) and "extra deps required only for testing" (ie catboost) if the original purpose of the file is for users to reference